app/parse.py

```diff
@@ -15,8 +26,93 @@ class Product:
     num_of_reviews: int
 
 
+def parse_single_product(soup: BeautifulSoup) -> Product:
+    title = soup.select_one(".title")["title"]
+    description = soup.select_one(".description").text.replace("\xa0", " ")
+    price = soup.select_one(".price").text.replace("$", "")
+    rating = soup.select(".ws-icon")
+    num_of_reviews = soup.select_one(".ratings > .review-count").text
+    return Product(
+        title=title,
+        description=description,
+        price=float(price),
+        rating=len(rating),
+        num_of_reviews=int(num_of_reviews.split()[0])
+    )
+
+
+def get_product_with_pagination_button(url: str) -> [Product]:
+    option = webdriver.ChromeOptions()
+    option.add_argument("headless")
+    driver = webdriver.Chrome(options=option)
+    driver.get(url)
+    while True:
+        try:
+            button_more = driver.find_element(
+                By.CLASS_NAME,
+                "ecomerce-items-scroll-more"
+            )
+            button_more.click()
+            time.sleep(1)
+        except ElementNotInteractableException:
+            break
+
+    page = driver.page_source
+    driver.close()
+    soup = BeautifulSoup(page, "html.parser")
+    products = soup.select(".product-wrapper")
+    return [parse_single_product(product) for product in products]
+
+
+def get_products_from_every_page(url: str) -> [Product]:
```

Comment:
The return type hint '[Product]' is not a valid syntax in Python. Use 'List[Product]' instead and make sure to import 'List' from 'typing'.